src/sentry/profiles/consumers/process/factory.py

```diff
@@ -0,0 +1,83 @@
+import logging
+from typing import Mapping
+
+import msgpack
+import sentry_sdk
+from arroyo.backends.kafka import KafkaProducer
+from arroyo.backends.kafka.consumer import KafkaPayload
+from arroyo.processing.strategies.abstract import ProcessingStrategy, ProcessingStrategyFactory
+from arroyo.processing.strategies.produce import ProduceAndCommit
+from arroyo.processing.strategies.run_task import RunTaskInThreads
+from arroyo.types import Commit, Message, Partition, Topic
+from django.conf import settings
+
+from sentry.models import Project
+from sentry.profiles.task import (
+    Profile,
+    _get_event_instance_for_legacy,
+    _get_event_instance_for_sample,
+    process_profile,
+)
+from sentry.utils import json, kafka_config
+
+logger = logging.getLogger(__name__)
+
+
+class ProcessProfileStrategyFactory(ProcessingStrategyFactory[KafkaPayload]):
+    """
+    This consumer processes replay recordings, which are compressed payloads split up into
+    chunks.
+    """
+
+    def process_message(self, message: Message[KafkaPayload]) -> KafkaPayload:
+        message_dict = msgpack.unpackb(message.payload.value, use_list=False)
+        profile = json.loads(message_dict["payload"])
```

Comment:
What about using rapidjson here? This has helped performance in some of our consumers.

Reply:
Great idea.